ChangeLog.d/psa-read-only-keys.txt

```diff
@@ -0,0 +1,5 @@
+Bugfix
+   * The PSA API no longer allows the creation or destruction of keys with a
+     read-only lifetime. The persistence level PSA_KEY_PERSISTENCE_READ_ONLY
```

Comment:
Why? “Key with a read-only lifetime” and “key with a read-only persistence level” are synonymous, but “read-only lifetime” is easier to understand.

Reply:
Because in the context of PSA crypto, "read-only" is a persistence level not a lifetime. A lifetime (in the context of PSA crypto) is the combination of a persistence level and a location. I understand that more generally "read-only lifetime" is completely fine but here I find "read-only persistence level" more precise than "read-only lifetime".

Reply:
A key is read-only if its lifetime is read-only. A lifetime is read-only if its persistence level is `PSA_KEY_PERSISTENCE_READ_ONLY`. “Read-only lifetime” is synonymous with “read-only persistence level”, but removing the word “lifetime” from this sentence makes it harder to understand.

Reply:
Nevermind, closing this conversation.